taurus_pyqtgraph/taurusmodelchoosertool.py

```diff
@@ -202,6 +205,10 @@ def __init__(self, parent=None):
         self.plot_item = None
         self.legend = None
         self._curveColors = None
+        self._parent = parent
+        if parent is not None:
+            # Filter drag and drop parent's event
+            parent.installEventFilter(self)
```

Comment:
For a tpg tool, the parent should not be checked or used in the constructor. Instead it should be delayed until  `attachToPlotItem` or later.

Reply:
> For a tpg tool, the parent should not be checked or used in the constructor. Instead it should be delayed until attachToPlotItem or _onTriggered.

I moved it to setParent method.